File: Python/7py 3/string_counter.py
"""
|--------------------------------------------------------------------------
| How many times does it contain?
|--------------------------------------------------------------------------
|
"""

def string_counter(string, char):
   res = 0
   for i in string:
      if i == char: res += 1
   return res


def string_counter(string, char):
   return string.count(char)


def string_counter(string, char):
   return sum(1 for i in string if char == i)


print(string_counter("Hello World!", "o"), 2)
print(string_counter("Do you like Harry Potter?", "?"), 1)
print(string_counter("abcdefg", "a"), 1)

File: PHP/7php 2/stringCounter.php
<?php
/*
|--------------------------------------------------------------------------
| How many times does it contain?
|--------------------------------------------------------------------------
|
*/

function stringCounter($inputS,$charS){
   return substr_count($inputS, $charS);
}


function stringCounter($inputS,$charS){
   preg_match_all('/(['.$charS.'])/', $inputS, $res);
   return count($res[0]);
}


function stringCounter(string $s, string $c): int {
   return count(array_filter(str_split($s), function ($e) use ($c) {
      return $c === $e;
   }));
}


print_r(stringCounter("Hello World!", "o") . ' ' . 2 . '
');
print_r(stringCounter("Do you like Harry Potter?", "?") . ' ' . 1 . '
');
print_r(stringCounter("abcdefg", "a") . ' ' . 1 . '
');